core/types/types.cc

```diff
@@ -655,8 +655,7 @@ bool LambdaParam::derivesFrom(const GlobalState &gs, SymbolRef klass) const {
 }
 
 bool SelfTypeParam::derivesFrom(const GlobalState &gs, SymbolRef klass) const {
-    Exception::raise(
-        "SelfTypeParam::derivesFrom not implemented, not clear what it should do. Let's see this fire first.");
+    return false;
```

Comment:
@XrXr WDYT about converting this into a user-visible error that this pattern isn't yet fully suported?
as-in, `ctx.beginError` dance.
This will make it clear that it doesn't work because the implementation isn't finalized.
The error can also suggest a workaround of `T.unsafe`-ing whatever expression had this type before.

Reply:
Hm, no, it actually doesn't work. You don't have location information here. :-(